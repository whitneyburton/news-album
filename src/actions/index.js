export const setNatGeo = articles => ({
  type: 'SET_NATIONAL_GEOGRAPHIC',
  articles
});

export const setCryptoCoins = articles => ({
  type: 'SET_CRYPTO_COINS',
  articles
});

export const setNewScientist = articles => ({
  type: 'SET_NEW_SCIENTIST',
  articles
});

export const toggleLoading = bool => ({
  type: 'TOGGLE_LOADING',
  bool
});

export const setError = message => ({
  type: 'SET_ERROR',
  message
});

export const setFavorites = article => ({
  type: 'SET_FAVORITES',
  article
});
