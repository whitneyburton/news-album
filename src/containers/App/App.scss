.App {
  padding: 20px 30px;
  font-family: Roboto, sans-serif;

  .App--NavLink-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin-bottom: 20px;

    .App--nav-btn {
      width: 30%;
      text-align: center;
      margin: auto 22px;
      text-decoration: none;
      color: #fff;
      font-weight: 300;
      letter-spacing: 2px;
      padding: 7px;
      border-radius: 12px;
      background-color: #4C3E3E;
    }
  }

  .App--quote-and-favorites {
    display: flex;
    justify-content: space-between;
    padding-bottom: 20px;
  }

  h1 {
    font-size: 3rem;
    text-align: right;
    margin: 0;
    padding-bottom: 10px;
    
    span {
      font-weight: 300;
    }
  }

  h2 {
    font-size: 2rem;
    margin-left: 20px;
    letter-spacing: 1px;
  }
  
  h3 {
    font-size: 1.5rem;
    text-align: right; 
    text-decoration: underline;
    
    span {
      font-weight: 300;
    }
  }
}
