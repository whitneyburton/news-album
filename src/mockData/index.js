export const mockNatGeoArticle = {
  author: 'Douglas Main',
  title: 'Why insect populations are plummeting',
  description: 'Sad insects.',
  url:
    'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
  urlToImage:
    'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
  publishedAt: '2019-02-19T02:07:15.6368398Z',
  id: 'b',
  isFavorite: false
};

export const mockCryptoCoinsArticle = {
  author: 'Greg Thompson',
  title: 'Ethereum Slayers',
  description: 'They are everywhere.',
  url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
  urlToImage:
    'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
  publishedAt: '2019-02-23T14:28:21Z',
  id: 'f',
  isFavorite: false
};

export const mockNewScientistArticle = {
  author: 'Leah Crane',
  title: 'Four stars full of burnt nuclear ash',
  description: 'Scary stars!',
  url:
    'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
  urlToImage:
    'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
  publishedAt: '2019-02-22T00:00:00Z',
  id: 'd',
  isFavorite: false
};

export const mockNatGeoArticles = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: false
  }
];

export const mockNatGeoArticlesWithFav = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: true
  }
];

export const mockNewScienceArticles = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: false
  }
];

export const mockNewScienceArticlesWithFav = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: true
  }
];

export const mockCryptoCoinsArticles = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: false
  }
];

export const mockCryptoCoinsArticlesWithFav = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: true
  }
];

export const mockNatGeoArticlesObj = {
  articles: [
    {
      author: 'Erica Johnson',
      title: '20 U.S. state parks',
      description: 'Pretty parks!',
      url:
        'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
      publishedAt: '2019-02-19T02:07:16.9805881Z',
      id: 'a',
      isFavorite: false
    },
    {
      author: 'Douglas Main',
      title: 'Why insect populations are plummeting',
      description: 'Sad insects.',
      url:
        'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
      publishedAt: '2019-02-19T02:07:15.6368398Z',
      id: 'b',
      isFavorite: false
    }
  ]
};

export const mockNewScienceArticlesObj = {
  articles: [
    {
      author: 'Chelsea Whyte',
      title: 'Wolves are spreading in Germany',
      description: 'Wolves everywhere!',
      url:
        'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
      publishedAt: '2019-02-23T00:00:00Z',
      id: 'c',
      isFavorite: false
    },
    {
      author: 'Leah Crane',
      title: 'Four stars full of burnt nuclear ash',
      description: 'Scary stars!',
      url:
        'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
      publishedAt: '2019-02-22T00:00:00Z',
      id: 'd',
      isFavorite: false
    }
  ]
};

export const mockCryptoCoinsArticlesObj = {
  articles: [
    {
      author: 'Joseph Young',
      title: 'Bitcoins spike',
      description: 'So much money!',
      url:
        'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
      publishedAt: '2019-02-23T14:45:23Z',
      id: 'e',
      isFavorite: false
    },
    {
      author: 'Greg Thompson',
      title: 'Ethereum Slayers',
      description: 'They are everywhere.',
      url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
      publishedAt: '2019-02-23T14:28:21Z',
      id: 'f',
      isFavorite: false
    }
  ]
};

export const mockNewSciArticlesStringified = {
  json: {
    articles: [
      {
        author: 'Chelsea Whyte',
        description: 'Wolves everywhere!',
        id: 'c',
        isFavorite: false,
        publishedAt: '2019-02-23T00:00:00Z',
        title: 'Wolves are spreading in Germany',
        url:
          'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg'
      },
      {
        author: 'Leah Crane',
        description: 'Scary stars!',
        id: 'd',
        isFavorite: false,
        publishedAt: '2019-02-22T00:00:00Z',
        title: 'Four stars full of burnt nuclear ash',
        url:
          'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg'
      }
    ]
  },
  ok: true
};
