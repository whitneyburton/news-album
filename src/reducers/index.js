import { combineReducers } from 'redux';
import { natGeoReducer } from './natGeoReducer';
import { newScientistReducer } from './newScientistReducer';
import { cryptoCoinsReducer } from './cryptoCoinsReducer';
import { errorReducer } from './errorReducer';
import { isLoadingReducer } from './isLoadingReducer';

export const rootReducer = combineReducers({
  natGeoArticles: natGeoReducer,
  newScientistArticles: newScientistReducer,
  cryptoCoinsArticles: cryptoCoinsReducer,
  error: errorReducer,
  isLoading: isLoadingReducer
});
