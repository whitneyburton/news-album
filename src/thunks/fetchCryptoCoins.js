import { setCryptoCoins, setError, toggleLoading } from '../actions';
import { fetchData } from '../utils/api';
import { apiKey } from '../utils/api-key';
import shortid from 'shortid';

export const fetchCryptoCoins = () => {
  return async (dispatch) => {
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(`https://newsapi.org/v2/top-headlines?sources=crypto-coins-news&apiKey=${apiKey}`);
      const result = await response.json();
      const articles = result.articles.map(article => {
        article.id = shortid.generate();
        article.isFavorite = false;
        return article;
      })
      dispatch(setCryptoCoins(articles));
      dispatch(toggleLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}