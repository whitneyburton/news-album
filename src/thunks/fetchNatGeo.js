import { setNatGeo, setError, toggleLoading } from '../actions';
import { fetchData } from '../utils/api';
import { apiKey } from '../utils/api-key';
import shortid from 'shortid';

export const fetchNatGeo = () => {
  return async (dispatch) => {
    const url = `https://newsapi.org/v2/top-headlines?sources=national-geographic&apiKey=${apiKey}`;
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(url);
      const result = await response.json();
      const articles = result.articles.map(article => {
        article.id = shortid.generate();
        article.isFavorite = false;
        return article;
      })
      dispatch(setNatGeo(articles));
      dispatch(toggleLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}