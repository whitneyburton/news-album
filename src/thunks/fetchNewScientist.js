import { setNewScientist, setError, toggleLoading } from '../actions';
import { fetchData } from '../utils/api';
import { apiKey } from '../utils/api-key';

export const fetchNewScientist = () => {
  return async (dispatch) => {
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(`https://newsapi.org/v2/top-headlines?sources=new-scientist&apiKey=${apiKey}`);
      const result = await response.json();
      dispatch(setNewScientist(result.articles));
      dispatch(toggleLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}