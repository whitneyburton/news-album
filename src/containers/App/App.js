import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter, Route, Switch } from 'react-router-dom';
import ArticleContainer from '../ArticleContainer/ArticleContainer';
import { Loader } from '../../components/Loader/Loader';
import { Popup } from '../Popup/Popup';
import { Home } from '../../components/Home/Home';
import { Nav } from '../../components/Nav/Nav';
import { Error404 } from '../../components/Error404/Error404';
import { fetchCryptoCoins } from '../../thunks/fetchCryptoCoins';
import { fetchNatGeo } from '../../thunks/fetchNatGeo';
import { fetchNewScientist } from '../../thunks/fetchNewScientist';

export class App extends Component {
  componentDidMount = () => {
    this.props.fetchNatGeo();
    this.props.fetchCryptoCoins();
    this.props.fetchNewScientist();
  }

  getArticleRoute = ({ match }) => {
    const { natGeoArticles, cryptoCoinsArticles, newScientistArticles } = this.props;
    const { id } = match.params;
    let currentArticle;
    if (match.path.includes('national')) {
      currentArticle = natGeoArticles.find(article => article.id === id);
    } else if (match.path.includes('crypto')) {
      currentArticle = cryptoCoinsArticles.find(article => article.id === id);
    } else if (match.path.includes('scientist')) {
      currentArticle = newScientistArticles.find(article => article.id === id);
    }
    return currentArticle ? ([
      <ArticleContainer match={match} isDisabled={true} />,
      <Popup currentArticle={currentArticle} match={match} />]) :
      <Error404 />;
  }

  render() {
    const { isLoading } = this.props;
    return (
      <div className="App">
        <h1>news<span>album</span></h1>
        <div className='App--quote-and-favorites'>
          <div>
            <h2 className='App--learn'>LEARN +</h2>
            <h2 className='App--get-inspired'>GET INSPIRED.</h2>
          </div>
          <h3>my<span>favorites</span></h3>
        </div>
        <Nav />
        {isLoading && <Loader />}
        <Switch>
          <Route
            path='/crypto-coins/:id'
            render={this.getArticleRoute} />
          <Route
            path='/national-geographic/:id'
            render={this.getArticleRoute} />
          <Route
            path='/new-scientist/:id'
            render={this.getArticleRoute} />
          <Route
            path='/national-geographic'
            render={({ match }) => <ArticleContainer match={match} />} />
          <Route
            path='/new-scientist'
            render={({ match }) => <ArticleContainer match={match} />} />
          <Route
            path='/crypto-coins'
            render={({ match }) => <ArticleContainer match={match} />} />
          <Route
            exact
            path='/'
            component={Home} />
          <Route component={Error404} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  natGeoArticles: state.natGeoArticles,
  newScientistArticles: state.newScientistArticles,
  cryptoCoinsArticles: state.cryptoCoinsArticles,
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchNatGeo: () => dispatch(fetchNatGeo()),
  fetchNewScientist: () => dispatch(fetchNewScientist()),
  fetchCryptoCoins: () => dispatch(fetchCryptoCoins())
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));

App.propTypes = {
  fetchNatGeo: PropTypes.func,
  fetchNewScientist: PropTypes.func,
  fetchCryptoCoins: PropTypes.func,
  natGeoArticles: PropTypes.array,
  newScientistArticles: PropTypes.array,
  cryptoCoinsArticles: PropTypes.array,
  isLoading: PropTypes.bool
}