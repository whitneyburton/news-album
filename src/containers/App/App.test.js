import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { createStore } from 'redux';
import { rootReducer } from '../../reducers';
import { shallow } from 'enzyme';
import { fetchNatGeo } from '../../thunks/fetchNatGeo';
import { fetchCryptoCoins } from '../../thunks/fetchCryptoCoins';
import { fetchNewScientist } from '../../thunks/fetchNewScientist';
import * as data from '../../mockData';
import * as actions from '../../actions';

const mockProps = {
  natGeoArticles: data.mockNatGeoArticles,
  newScientistArticles: data.mockNewScienceArticles,
  cryptoCoinsArticles: data.mockCryptoCoinsArticles,
  isLoading: false,
  fetchNatGeo: jest.fn(),
  fetchNewScientist: jest.fn(),
  fetchCryptoCoins: jest.fn(),
  setNatGeo: jest.fn(),
  setNewScientist: jest.fn(),
  setCryptoCoins: jest.fn()
};

const mockMatch = { params: { id: 'a' }, path: 'national-geographic/a' };

jest.mock('../../thunks/fetchCryptoCoins.js');
jest.mock('../../thunks/fetchNatGeo.js');
jest.mock('../../thunks/fetchNewScientist.js');

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App {...mockProps} match={mockMatch} />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = createStore(rootReducer);
    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <App {...mockProps} />
        </BrowserRouter>
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when isLoading is true', () => {
    wrapper = shallow(<App {...mockProps} isLoading={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchNatGeo, fetchCryptoCoins, and fetchNewScientist on componentDidMount', () => {
    wrapper.instance().componentDidMount();
    expect(mockProps.fetchNatGeo).toHaveBeenCalled();
    expect(mockProps.fetchCryptoCoins).toHaveBeenCalled();
    expect(mockProps.fetchNewScientist).toHaveBeenCalled();
  });

  describe('getArticleRoute', () => {
    it('should return the ArticleContainer and Popup components when there is a NAT GEO currentArticle', () => {
      const result = wrapper.instance().getArticleRoute({ match: mockMatch });
      expect(result).toHaveLength(2);
    });

    it('should return the ArticleContainer and Popup components when there is a NEW SCIENTIST currentArticle', () => {
      const mockMatch = { params: { id: 'c' }, path: 'new-scientist/c' };
      const result = wrapper.instance().getArticleRoute({ match: mockMatch });
      expect(result).toHaveLength(2);
    });

    it('should return the ArticleContainer and Popup components when there is a CRYPTO COINS currentArticle', () => {
      const mockMatch = { params: { id: 'e' }, path: 'crypto-coins/e' };
      const result = wrapper.instance().getArticleRoute({ match: mockMatch });
      expect(result).toHaveLength(2);
    });

    it('should return the Error404 component when there is no currentArticle match', () => {
      const mockMatch = { params: { id: 'z' }, path: 'national-geographic/z' };
      const result = wrapper.instance().getArticleRoute({ match: mockMatch });
      const errorWrapper = shallow(result);
      expect(errorWrapper.find('.Error404')).toHaveLength(1);
    });
  });

  describe('getFavoritesFromStorage', () => {
    it('should call setNatGeo when there is a national geographic article in storage', () => {
      localStorage.setItem(
        'favorites',
        JSON.stringify(['Why insect populations are plummeting'])
      );
      wrapper.instance().getFavoritesFromStorage();
      expect(mockProps.setNatGeo).toHaveBeenCalled();
    });

    it('should call setNewScientist when there is a New Scientist article in storage', () => {
      localStorage.setItem(
        'favorites',
        JSON.stringify(['Four stars full of burnt nuclear ash'])
      );
      wrapper.instance().getFavoritesFromStorage();
      expect(mockProps.setNewScientist).toHaveBeenCalled();
    });

    it('should call setCryptoCoins when there is a National Geographic article in storage', () => {
      localStorage.setItem('favorites', JSON.stringify(['Ethereum Slayers']));
      wrapper.instance().getFavoritesFromStorage();
      expect(mockProps.setCryptoCoins).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should return a props object with the 3 article types and isLoading', () => {
      const mockState = {
        natGeoArticles: data.mockNatGeoArticles,
        newScientistArticles: data.mockNewScienceArticles,
        cryptoCoinsArticles: data.mockCryptoCoinsArticles,
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a fetchNatGeo thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchNatGeo();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchNatGeo();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a fetchNewScientist thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchNewScientist();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchNewScientist();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a fetchCryptoCoins thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchCryptoCoins();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchCryptoCoins();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a setNewScientist action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.setNewScientist(
        data.mockNewScienceArticles
      );
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setNewScientist(data.mockNewScienceArticles);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a setCryptoCoins action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.setCryptoCoins(
        data.mockCryptoCoinsArticles
      );
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCryptoCoins(data.mockCryptoCoinsArticles);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
