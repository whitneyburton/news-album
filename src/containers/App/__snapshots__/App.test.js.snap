// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Nav />
  <Switch>
    <Route
      path="/crypto-coins/:id"
      render={[Function]}
    />
    <Route
      path="/national-geographic/:id"
      render={[Function]}
    />
    <Route
      path="/new-scientist/:id"
      render={[Function]}
    />
    <Route
      path="/favorites/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/national-geographic"
    />
    <Route
      component={[Function]}
      path="/new-scientist"
    />
    <Route
      component={[Function]}
      path="/crypto-coins"
    />
    <Route
      component={[Function]}
      path="/favorites"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <p
    className="attribution"
  >
    Powered by
     
    <a
      className="attribution-link"
      href="https://newsapi.org/"
    >
      News API
    </a>
  </p>
</div>
`;

exports[`App should match the snapshot when isLoading is true 1`] = `
<div
  className="App"
>
  <Header />
  <Nav />
  <Loader />
  <p
    className="attribution"
  >
    Powered by
     
    <a
      className="attribution-link"
      href="https://newsapi.org/"
    >
      News API
    </a>
  </p>
</div>
`;
